pkg/object/object.go

```diff
@@ -157,11 +205,22 @@ func (c *Container) validate() error {
 			return fmt.Errorf("failed to validate mount: %v", err)
 		}
 
+		var mountType *string
+		if mount.VolumeRef != nil {
+			mountType = mount.VolumeRef
+		}
+		if mount.SecretRef != nil {
+			mountType = mount.SecretRef
+		}
+
 		// mountPath should not collide, which means you should not do multiple mounts in same place
 		if v, ok := allMounts[mount.MountPath]; ok {
-			return fmt.Errorf("mount %q: mountPath %q: cannot have same mountPath as %q", mount.VolumeRef, mount.MountPath, v)
+			return fmt.Errorf("mount %q: mountPath %q: cannot have same mountPath as %q", *mountType, mount.MountPath, v)
 		}
-		allMounts[mount.MountPath] = mount.VolumeRef
+		allMounts[mount.MountPath] = *mountType
+
+		// validate volumeSubPath
+		// TODO: if there is someway to do it
```

Comment:
these two unnecessary comments have been added, it was already moved to `mount.validate()` ?